### ISSUE TITLE
Profession Level up on Gray Recipe

### ISSUE THREAD

**Issue by Cosmic-Cocoa** [chromiecraft/chromiecraft#155]
##### ISSUE: FACTION SIDE:

![Alliance](https://user-images.githubusercontent.com/1884642/108204869-3a88d100-711c-11eb-8179-e1b9b73ed450.png) & ![Horde](https://user-images.githubusercontent.com/1884642/108204991-63a96180-711c-11eb-9ed9-138c233070a6.png)

##### CONTENT PHASE:

Any phase, I'd think.

##### SMALL DESCRIPTION:

I earned a profession skill level from a gray recipe, namely Minor Inscription Research.
![Profession Increase on Gray Recipe](https://user-images.githubusercontent.com/78349628/110222038-28858d00-7e95-11eb-9c5c-07a929cd20e8.png)

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Gray recipes should never give a skill increase.
[Wowwiki Page](https://wowwiki-archive.fandom.com/wiki/Gray#Gray_Recipes)

I saw a lot of other posts that said as much, but mostly as comments on wowhead and profession guides.

##### CURRENT BEHAVIOUR:

I did my Minor Inscription Research today and got a skill up. This is the first time I've noticed it and that recipe has been gray for me for quite a while. Not sure if this would impact other, non-cooldown, recipes. I haven't noticed any when making inks or anything.

##### STEPS TO REPRODUCE THE PROBLEM:

Step 1. Give yourself inscription. 
Step 2. Remove the cooldown for professions.
Step 3. Perform several Minor Inscription Research and see if you get any skill ups. 

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/c2f274e06dcf85c6ccc0ab282dd6f23e8ce49c8f

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

**Comment by Stifler82** [comment]
I actually knew that gray skills have a very low chance of making a skill point but the percentage is really low, so low that it can be considered a rare event.

**Comment by Cosmic-Cocoa** [comment]
Oh, I had never heard that. If that's the case, then I guess it's not a bug and I just got lucky. Apologies.

**Comment by Stifler82** [comment]
```
#    SkillChance.Orange
#    SkillChance.Yellow
#    SkillChance.Green
#    SkillChance.Grey
#        Description: Chance to increase skill based on recipe color.
#        Default:     100 - (SkillChance.Orange)
#                     75  - (SkillChance.Yellow)
#                     25  - (SkillChance.Green)
#                     0   - (SkillChance.Grey)

SkillChance.Orange = 100
SkillChance.Yellow = 75
SkillChance.Green  = 25
SkillChance.Grey   = 0
```
All I can tell you is that our config is configured like this..

**Comment by Stifler82** [comment]
> Oh, I had never heard that. If that's the case, then I guess it's not a bug and I just got lucky. Apologies.

@Cosmic-Cocoa 
But I'm not sure about this..

**Comment by Stifler82** [comment]
@Cosmic-Cocoa 
probably that "0" we have set is not a true "0" it will probably be a "0.001" so as to create the "rare event".

**Comment by r-o-b-o-t-o** [comment]
I think it really is a bug and we should reopen this.
I had this happen to me a lot the last couple of weeks on the Minor Inscription Research:

* may 19th:
![image](https://user-images.githubusercontent.com/24357633/120346576-c86b7f00-c2fb-11eb-816e-a30ce85398c8.png)

* may 21st:
![image](https://user-images.githubusercontent.com/24357633/120346756-f2bd3c80-c2fb-11eb-90e6-ee42d566b997.png)

* may 24th:
![image](https://user-images.githubusercontent.com/24357633/120346826-05d00c80-c2fc-11eb-8ad4-f344915ef58e.png)

* may 25th:
![image](https://user-images.githubusercontent.com/24357633/120346916-1b453680-c2fc-11eb-9299-8f90b47c267d.png)

* may 26th:
![image](https://user-images.githubusercontent.com/24357633/120347015-31eb8d80-c2fc-11eb-856f-650976e6d110.png)

* may 28th:
![image](https://user-images.githubusercontent.com/24357633/120347095-429c0380-c2fc-11eb-9b17-6c15746f54e9.png)

* june 1st:
![image](https://user-images.githubusercontent.com/24357633/120347218-5f383b80-c2fc-11eb-995b-4f8f16d9c560.png)

I don't think it should even be a "really rare event", unless someone finds a source for this, but anyway it happened to me with a 50% rate if you count the dates, so it's not even close to rare...

**Comment by necropola** [comment]
I thought the special case here is that you **learned a new Minor Glyph** upon doing **Minor Inscription Research** which itself is gray. At least I got a skill point every time I discovered a new glyph on ChromieCraft, even after the research itself turned gray. I think I even got 2 points in the beginning (one for the research which was still yellow and one for the discovery):

Wasn't that actually a thing on retail, too? I rermember vaguely that this was a way to slowly but cheaply level up Inscription. But probably only until level 125 when I understand the historic article on Wowpedia correctly: https://wowpedia.fandom.com/wiki/Minor_Inscription_Research?oldid=2382884
> The research doesn't go yellow until 125, so it is far and away the cheapest way to level inscription, though the cooldown makes it the slowest as well.

So I agree. This is a bug. Discovering a new glyph should **not** by itself grant a skill point. The chance for a skill point should entirely be based on the research being yellow, green or gray.
